### ISSUE TITLE
[Prims+NVFuser] Aten2Aten decomp hurting performance

### ISSUE THREAD

**Issue by SherlockNoMad** [pytorch/pytorch#80966]

**Comment by ngimel** [comment]
Yeah `native_layer_norm_backward`, like `var_mean`, should not be decomposed for nvfuser and should instead be directly sent to it, see discussion in #80187

**Comment by SherlockNoMad** [comment]
> Yeah `native_layer_norm_backward`, like `var_mean`, should not be decomposed for nvfuser and should instead be directly sent to it, see discussion in #80187 And `embedding_dense_backward` shouldn't be sent to nvfuser at all, nvfuser doesn't support it.

This implies a few things
- we need to maintain a backend-specific blocklist (or allowlist) for aten2aten decomp
- Eager lowering pipeline need an API to accept custom decomp_tabble 
- We need a way to test the performance impact of aten2aten decomp, and catch the cases where nvFuser doesn't handle.

**Comment by ngimel** [comment]
> we need to maintain a backend-specific blocklist (or allowlist) for aten2aten decomp

Correct, backends have different capabilities and should be able to opt into some non-decomposed operations
>Eager lowering pipeline (AoTAutograd) need an API to accept custom decomp_table

AOT already accepts custom decomposition list - is that enough?

>We need a way to test the performance impact of aten2aten decomp, and catch the cases where nvFuser doesn't handle.

Correct

**Comment by davidberard98** [comment]
As I understand it the original plan was to have:
1) aten to canonical-aten decomp, which would be one of the first passes
2) canonical-aten to prim decomp, which happens _after_ partitioning and where the backend might choose not to decompose some of the minimal-aten ops into prims.

But this suggests that we can't run always the aten to canonical-aten decomp, since we need to worry about which backend is using it. In this case, is there any use case at all for aten to aten decompositions?

Or alternatively, do we want to be more careful about what decompositions are part of the set of "canonical-aten" ops?

**Comment by ezyang** [comment]
There's a parallel discussion going on here: https://github.com/pytorch/pytorch/issues/80541 I think my speculation proposal would also help with the issue here

**Comment by IvanYashchuk** [comment]
There're two distinct ways to reach nvFuser: through TorchScript and through primTorch. And these two ways shouldn't be assumed the same or equivalent (https://github.com/pytorch/pytorch/pull/79439#discussion_r903377070), https://github.com/pytorch/pytorch/pull/79439#discussion_r903368723).

Ops like `torch.ops.aten.native_layer_norm_backward` should be sent to nvFuser through TorchScript (`torch.jit.script`) then it would use nvFuser's internal decomposition and fusion. Before sending through the primtorch way, is this decomposition used
https://github.com/pytorch/pytorch/blob/1d90d6ee60f2e1850556fd26d7ffde950ab992a1/torch/_decomp/decompositions.py#L882-L954
are calls to `torch.` translated to `torch._refs`? Are these `refs` decomposed further to make a graph consisting solely of `torch.ops.prims`?

Prims+nvFuser would work only for these prims `torch.ops.prims.OP_NAME.impl_nvfuser is not None` (https://github.com/pytorch/pytorch/issues/80541#issuecomment-1176243956) and nothing else.

**Comment by jjsjann123** [comment]
Should we close this bug now, since we are now only running a subset of aten2aten decompositions at the moment.
